sqlstore/oracle_spec_test.go

```diff
@@ -380,7 +386,115 @@ func testOracleSpecPaginationLastBefore(t *testing.T) {
 	specs := createOracleSpecPaginationTestData(t, ctx, bs, os)
 	last := int32(3)
 	before := specs[7].Cursor().Encode()
-	pagination, err := entities.NewCursorPagination(nil, nil, &last, &before)
+	pagination, err := entities.NewCursorPagination(nil, nil, &last, &before, false)
+	require.NoError(t, err)
+
+	got, pageInfo, err := os.GetSpecsWithCursorPagination(ctx, "", pagination)
+	require.NoError(t, err)
+
+	assert.Equal(t, specs[4:7], got)
+	assert.Equal(t, entities.PageInfo{
+		HasNextPage:     true,
+		HasPreviousPage: true,
+		StartCursor:     specs[4].Cursor().Encode(),
+		EndCursor:       specs[6].Cursor().Encode(),
+	}, pageInfo)
+}
+
+func testOracleSpecPaginationNoPaginationNewestFirst(t *testing.T) {
```

Comment:
looks good, one thought, we repeat the pagination tests for each type,  which I guess means that we are really just testing the pagination logic multiple times, though there is some type specific stuff for each type, I wondered if each type needs to do the full test suite for pagination or if a more limited test for each type would suffice?

Reply:
I was afraid that there weren't enough tests for it to be honest. I think the 5 test cases I had were the minimum:

- No Pagination
- First/Last
- First/Last with Cursor

They're only doubled up because now we have sorting order from newest->oldest and vice versa. The cursors for each data-type can be quite different and the test have been helpful to make sure that I have constructed them properly in order to get the results we expect. 

I think in the long run, the pagination could get more complicated if they ask for ordering by specific columns which is something that has come up in discussions I've seen on Slack.